dex/farm/src/lib.rs

```diff
@@ -112,6 +112,8 @@ pub trait Farm:
         &self,
         #[var_args] opt_accept_funds_func: OptionalArg<ManagedBuffer>,
     ) -> SCResult<EnterFarmResultType<Self::Api>> {
+        let apr_multiplier = self.locked_rewards_apr_multiplier().get();
+        require!(apr_multiplier > 1, "farm does not support locked rewards");
```

Comment:
I think that it is better to make a completely new set of contracts - one which generates locked rewards - thus metaESDTs and one which generates with ESDTs. That will make the overall contracts smaller.